lib/index.js

```diff
@@ -447,13 +447,34 @@ Simulcast.prototype.mungeRemoteDescription = function (desc) {
             implodeRemoteSimulcast(mLine);
         }
 
-        // If native simulcast is enabled, we must append the x-goog-conference
-        // attribute to the SDP.
-        if (self.ssrcCache.length < 1) {
-            removeGoogConference(mLine);
-        } else {
+        // Make sure that we ALWAYS add the conference flag to the remote
+        // description: That flag is deprecated but, as of this writing
+        // (09/2019), it's still used in jitsi-meet to enable legacy simulcast
+        // when screen-sharing (https://cs.chromium.org/chromium/src/third_party/webrtc/media/base/media_channel.h?rcl=f2773b54647633d5725fa25dea883f99dd1b400c&l=838).
+        //
+        // We've observed the following chain of events that almost always
+        // results in connectivity issues at the receiver:
+        //
+        // 1. The user joins a call with startVideoMuted=true, so there's no
+        //    jitsi track created.
+        // 2. The user enables screen-sharing. At this point we do a
+        //    renegotiation (which starts with an sRD); the ssrcCache is empty,
+        //    so the conference flag is not added to the remote description SDP.
+        // 3. The next step of the re-negotiation is to set the answer, so
+        //    mungeLocalDescription is called, and because now we have a track
+        //    due to the user enabling screen-sharing, it enables simulcast.
+        // 4. sLD is called with simulcast ssrcs, but the sRD call of step 2
+        //    didn't add the conference flag in the SDP, so the screen-sharing
+        //    is without simulcast.
+        // 5. Any subsequent re-negotiation (for example, due to a participant
+        //    joining/leaving the call) will enable simulcast and this switch
+        //    almost always results in a broken stream at the receiver.
+
+        //if (self.ssrcCache.length < 1) {
```

Comment:
Generally speaking, I'm not a big fan of leaving commented out code in, git history has it anyway, and your comment does an excellent job at explaining.

Reply:
thanks Saul, you're right, I removed the commented code.